learn/advanced/asynchronous_operations.md

```diff
@@ -0,0 +1,142 @@
+| `uid`        | integer | The unique sequential identifier of the task                                                                     |
+| `indexUid`   | string  | The unique index identifier                                                                                      |
+| `status`     | string  | The status of the task, possible values are `enqueued`, `processing`, `succeeded`, `failed`                                                                                                                                    |
+| `type`       | string  | The type of task, possible values are `indexCreation`, `indexUpdate`, `indexDeletion`, `documentsAddition`, `documentsPartial`, `documentsDeletion`, `settingsUpdate`, `clearAll`                                                                       |
+| `details`    | object  |  Contains detailed information on the task payload                                                               |
+| `error`      | object  | Shows error details and context when a task has a `failed` status                                                |
+| `duration`   | string  | Represents the total elapsed time the engine was in the `processing` state in the ISO 8601 duration format     |
+| `enqueuedAt` | string  | Represents the date and time in the ISO 8601 format when the task has been `enqueued`                          |
+| `startedAt`  | string  | Represents the date and time in the ISO 8601 format when the task has been dequeued and started being processed. The default is `null`.                                                                                                                      |
+| `finishedAt` | string  | Represents the date and time in the ISO 8601 format when the task has the `failed` or `succeeded` status. The default is `null`.                                                                                                                          |
+
+If a task fails due to an error, all error fields will be appended to the response.
+
+### Task `status`
+
+Task responses always contain a field indicating the request's current `status`. This field can have one of four possible values:
+
+- `enqueued`: the task request has been received and will be processed soon
+- `processing`: the task is being processed
+- `processed`: the task has been successfully processed
+- `failed`: a failure occurred when processing the task
+
+The **final status of a `processed` task is `succeeded` or `failed`.**
+
+### Examples
+
+Suppose you add a new document to your instance using the [documents API endpoint](/reference/api/documents.md#add-or-replace-documents) and receive a `uid`.
+
+When you query the task endpoint using this id, you see that it has been enqueued:
+
+```json
+{
+    "uid": 1,
+    "indexUid": "movies",
+    "status": "enqueued",
+    "type": "documentsAddition",
+    "details": { "numberOfDocuments": 67493},
```

Comment:
Hi @maryamsulemani97 👋 

By default, a details field that can have a real value when the task is failed or succeeded will have a `null` value for `enqueued` and `processing` status.

As for `indexedDocuments` field in this example since the `documentsAddition` is `enqueued`.

Reply:
Thanks for the update!